tests/unit/Entity/ItemIdTest.php

```diff
@@ -84,10 +84,25 @@ public function testSerialize() {
 		$this->assertSame( '["item","Q1"]', $id->serialize() );
 	}
 
-	public function testUnserialize() {
+	/**
+	 * @dataProvider jsonProvider
+	 */
+	public function testUnserialize( $json, $expected ) {
 		$id = new ItemId( 'Q1' );
-		$id->unserialize( '["item","Q2"]' );
-		$this->assertSame( 'Q2', $id->getSerialization() );
+		$id->unserialize( $json );
+		$this->assertSame( $expected, $id->getSerialization() );
+	}
+
+	public function jsonProvider() {
```

Comment:
There is already an idSerializationProvider above. I wanted to avoid confusion. A possible alternative is phpSerializationProvider, because this is about PHP's Serialization interface. Is this ok for you?